### ISSUE TITLE
[Feature request] Include class names from .css as valid custom classes

### ISSUE THREAD

**Issue by adamlindqvist** [francoismassart/eslint-plugin-tailwindcss#24]
**Is your feature request related to a problem? Please describe.**
For every custom class included in a tailwind.css file, It is needed to add them to the whitelist option for the `no-custom-classname` rule. For projects where custom classes are used a lot, this could be hard to keep track off.

```css
/* tailwind.css */
@tailwind base;
@tailwind components;
@tailwind utilities;

@layer components {
    .btn {
        @apply p-4 border rounded
    }
   ...
}

```
**Describe the solution you'd like**
It would be nice if this plugin would include all classes created in the .css file  by default. 
Perhaps in addition to whitelisting specific classes, it could also accept paths for .css files with classes that should be whitelisted.

**Describe alternatives you've considered**
Using the whitelist option for the `no-custom-classname` rule.

**Comment by francoismassart** [comment]
That would require using a 3rd party CSS parser which "understands" Tailwind CSS's directive such as `@layer`...

Do you have any recommendations ?
I'll do some researches about his one.

**Comment by adamlindqvist** [comment]
I've not investigated this in detail at all, but the [tailwindcss-intellisense](https://github.com/tailwindlabs/tailwindcss-intellisense) might be a nice reference. 

For example: https://github.com/tailwindlabs/tailwindcss-intellisense/blob/master/src/lib/extractClassNames.ts to process css-files and get class names.

**Comment by francoismassart** [comment]
Indeed I found out about the `postcss-selector-parser` 👍 
You can also find other candidates on https://astexplorer.net/
![image](https://user-images.githubusercontent.com/704026/120152726-1eea9780-c1ee-11eb-871f-57a1fd372f1a.png)

What should be the strategy for choosing the CSS files to be parsed ?

- `tailwind.css` off course

but possibly all the css files under the project root folder...

- e.g. `/src/components/button/button.module.css`
- e.g. `/src/components/menu/menu.module.css`
- ...

What do you think ?

**Comment by adamlindqvist** [comment]
> Indeed I found out about the `postcss-selector-parser` 👍
> You can also find other candidates on https://astexplorer.net/
> ![image](https://user-images.githubusercontent.com/704026/120152726-1eea9780-c1ee-11eb-871f-57a1fd372f1a.png)
> 
> What should be the strategy for choosing the CSS files to be parsed ?
> 
> * `tailwind.css` off course
> 
> but possibly all the css files under the project root folder...
> 
> * e.g. `/src/components/button/button.module.css`
> * e.g. `/src/components/menu/menu.module.css`
> * ...
> 
> What do you think ?

Looks good to me 💯

**Comment by thebuilder** [comment]
It would make sense to have it as a user defined `glob`, so it doesn't just happen automatically. It should also not include `.module.css`, since those files are imported as JS objects inside the code.

**Comment by francoismassart** [comment]
@thebuilder, do you mean that the fact that the `.module.css` files are scoped their content should not be used to whitelist valid classnames **globally** ?
I think the `.module.css` files could still be used to declare classnames with `@apply`. 
I wonder how to make sure it does not pollute the whitelist for all other contextes.

**Comment by thebuilder** [comment]
Even if you use `@apply` in a CSS module, it will still be outputted in JS generated/hashed classname - So yeah, don't think it makes sense. Or am I missing something?

**Comment by francoismassart** [comment]
@thebuilder & @adamlindqvist let me know what you think about the PR !
First I did attempt to make the loading/parsing of css file asynchronous but from what I understood eslint rules must be synchronous by design.

**Comment by thebuilder** [comment]
@francoismassart regarding the default CSS filter list, I think it should be expanded a bit more:

* Exclude build output
  * Ignore `dist`
  * Ignore `*.min.css`

Is it possible to get the ignore list from `.eslintignore`? That would serve as a good base.
